src/design-pages/idoverride_redesign.rst

```diff
@@ -0,0 +1,400 @@
+.. _User information is being removed from SSSD local Override database: https://bugzilla.redhat.com/show_bug.cgi?id=1452883
+
+
+Problem statement
+*****************
+The original design for id-overrides is described in
+https://www.freeipa.org/page/V4/Migrating_existing_environments_to_Trust and
+all the features described there should of course work as expected after the
+redesign as well. Only the implementation should be updated with performance
+and reliability in mind.
+
+Initial the id-overrides were planned as a feature for the IPA provider. Some
+time later the local id-overrides were added keeping the way how id-overrides
+were handled by the ipa provider. This meant that when the id-override was
+added the related user was looked up as well and added to the cache together
+with the id-override data. And if the user was removed from the cache for
+whatever reasons the matching id-override was removed from the cache as well.
+This works well for well for IPA since there is a central storage for all
+current id-overrides on the IPA servers. But the local id-override data is
+lost and if the user is added to the cache again there will be no id-override
+data to apply. So it would make sense to decouple the id-override data in
+SSSD's cache form the actual user and group objects.
+
+The id-override data is typically stored in a separate tree/container in the
+cache. There is an exception for the "Default Trust View" on IPA servers where
+the id-override data is directly applied to the user or group objects in the
+cache (see original design for the reason) but with the new design it should be
+possibel to get rid of this special handling. So instead of removing the whole
+id-override object automatically only the back-reference to the user or group
+object can be removed and the id-override object is only removed if the source
+object is removed or the local id-override is explicitly removed.
+
+This decoupling would also allow to handle the different change times of
+id-overrides, user and group objects more efficiently. While changes to groups
+are expected to happen quite often by adding or removing member, changes to
+user objects should be less often. Id-overrides are typically created once and
+will most probably never change. Currently, to look a user, e.g. by name,
+there will be first and LDAP search for an id-override with the given name.
+Depending on if an id-override was found or not the user is looked up by the
+id-override anchor or by name respectively. If no id-override was found before
+another search for the id-override is done with the anchor to see if other
+attributes than the name is overwritten. For users or group with no
+id-overrides this means three LDAP searches.
+
+If the id-override data is downloaded once into the cache this can be handled
+more efficiently because the id-override data is already present on the local
+client and can be kept up-to-data in the similar scheme as sudo rules. For
```

Comment:
Do you mean in the same scheme as sudo rules or in **a** similar scheme as the sudo rules?